src/main/java/org/jvnet/hudson/test/JenkinsSessionRule.java

```diff
@@ -55,6 +55,15 @@
      */
     private int port;
 
+    /**
+     * Get the Jenkins home directory, which is consistent across restarts.
+     * Normally it will suffice to use {@link LocalData} to populate files.
+     */
+    public File getHome() {
+        assert home != null : "JENKINS_HOME has not been allocated yet";
```

Comment:
Probably better made into an `IllegalStateException`, though assertions ought to be enabled in tests IIRC.

Reply:
Yeah, I had it that way at first, but from some quick testing was not able to trigger the condition in normal usage so I just used an assertion for brevity. I can switch back.

Reply:
8f5d840